### PR TITLE
fix typo in `specificity_at_sensitivity`

### PR DESCRIPTION
**Pull request by oguz-hanoglu**
## What does this PR do?

- specicity used instead of specificity -> Fixed.
- The issue was: init.py has duplicates (class based one), init.py's are not organized in the same order (functional and class based) 
- The issue addressed as removing duplicates and reorganizing the list order.

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

- [X] Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2199.org.readthedocs.build/en/2199/

### CODE REVIEW

**Review by Borda — APPROVED** [review]
it si breaking change but worse to fix it

**Comment by oguz-hanoglu** [comment]
@Borda  To avoid any misunderstandings among other reviewers, could you please clarify your comment? I assume you mean: "It is not a breaking change, but it is worth fixing." Am I correct?

File: src/torchmetrics/functional/classification/specificity_sensitivity.py [hunk 1]
```diff
@@ -403,7 +403,7 @@ def multilabel_specificity_at_sensitivity(
     return _multilabel_specificity_at_sensitivity_compute(state, num_labels, thresholds, ignore_index, min_sensitivity)
 
 
-def specicity_at_sensitivity(
+def specificity_at_sensitivity(
```
Comment (Borda) [inline src/torchmetrics/functional/classification/specificity_sensitivity.py:406]: this is rename public function, so someone was using it in his project; this change will break his code

Comment (oguz-hanoglu) [reply src/torchmetrics/functional/classification/specificity_sensitivity.py:406]: I see. Then do you think this rename operation is still worth to do, or is it better to keep it as "specicity"?

Comment (Borda) [reply src/torchmetrics/functional/classification/specificity_sensitivity.py:406]: cc: @lantiga

**Review by matsumotosan — APPROVED** [review]

File: src/torchmetrics/functional/classification/specificity_sensitivity.py (hunk 1 above)
Comment (Borda) [reply src/torchmetrics/functional/classification/specificity_sensitivity.py:406]: With adding come compatibility path and deprecation warning, all shall be smooth...

Comment (lantiga) [reply src/torchmetrics/functional/classification/specificity_sensitivity.py:406]: Yep let's add the new one, keep the old one around and add the deprecation warning.

Comment (lantiga) [reply src/torchmetrics/functional/classification/specificity_sensitivity.py:406]: Thanks for the fix @oguz-hanoglu

Comment (Borda) [reply src/torchmetrics/functional/classification/specificity_sensitivity.py:406]: @oguz-hanoglu could you pls follow examples with deprecations and set it for removal in 2.0

**Review by Borda — CHANGES_REQUESTED** [review]

File: src/torchmetrics/functional/classification/specificity_sensitivity.py (hunk 1 above)
Comment (oguz-hanoglu) [reply src/torchmetrics/functional/classification/specificity_sensitivity.py:457]: @Borda I added the old method back and added a deprecation warning.

File: src/torchmetrics/functional/classification/specificity_sensitivity.py [hunk 2]
```diff
@@ -414,7 +416,56 @@ def specicity_at_sensitivity(
...
+        return multiclass_specificity_at_sensitivity(  # type: ignore
+            preds, target, num_classes, min_sensitivity, thresholds, ignore_index, validate_args
+        )
+    if task == ClassificationTask.MULTILABEL:
+        if not isinstance(num_labels, int):
+            raise ValueError(f"`num_labels` is expected to be `int` but `{type(num_labels)} was passed.`")
+        return multilabel_specificity_at_sensitivity(  # type: ignore
+            preds, target, num_labels, min_sensitivity, thresholds, ignore_index, validate_args
+        )
+    raise ValueError(f"Not handled value: {task}")
```
Comment (Borda) [inline src/torchmetrics/functional/classification/specificity_sensitivity.py:454]:
```suggestion
    return specificity_at_sensitivity(
    preds=preds,
    target=target,
    task=task,
    min_sensitivity=min_sensitivity,
    thresholds=thresholds,
    num_classes=num_classes,
    num_labels=num_labels,
    ignore_index=ignore_index,
    validate_args=validate_args
    )
```

**Review by SkafteNicki — APPROVED** [review]

File: src/torchmetrics/functional/classification/specificity_sensitivity.py [hunk 3]
```diff
@@ -414,7 +416,49 @@ def specicity_at_sensitivity(
...
+    This is done by first calculating the Receiver Operating Characteristic (ROC) curve for different thresholds and
+    the find the specificity for a given sensitivity level.
+
+    This function is a simple wrapper to get the task specific versions of this metric, which is done by setting the
+    ``task`` argument to either ``'binary'``, ``'multiclass'`` or ``multilabel``. See the documentation of
+    :func:`~torchmetrics.functional.classification.binary_specificity_at_sensitivity`,
+    :func:`~torchmetrics.functional.classification.multiclass_specificity_at_sensitivity` and
+    :func:`~torchmetrics.functional.classification.multilabel_specificity_at_sensitivity` for the specific details of
+    each argument influence and examples.
+
```
Comment (Borda) [inline src/torchmetrics/functional/classification/specificity_sensitivity.py:430]: let's replace this with deprecation  warning

File: src/torchmetrics/functional/classification/specificity_sensitivity.py [hunk 4]
```diff
@@ -403,6 +404,7 @@ def multilabel_specificity_at_sensitivity(
     return _multilabel_specificity_at_sensitivity_compute(state, num_labels, thresholds, ignore_index, min_sensitivity)
 
 
+# create specicity_at_sensitivity that calls specificity_at_sensitivity
```
Comment (Borda) [inline src/torchmetrics/functional/classification/specificity_sensitivity.py:407]:
```suggestion
```

**Review by Borda — APPROVED** [review]

**Pull request merged**